File: ifBanjo/ifBanjo.js
function areYouPlayingBanjo(name) {
  /*
  Create a function which answers the question "Are you playing banjo?".
  If your name starts with the letter "R" or lower case "r", you are playing banjo!
  
  The function takes a name as its only argument, and returns one of the following strings:
  
  name + " plays banjo" 
  name + " does not play banjo"
  Names given are always valid strings.
    */
  if (name[0] === 'R' || name[0] === 'r') {
    return name + ' plays banjo';
  }
  return name + ' does not play banjo';
}

// alternative solns
function areYouPlayingBanjo(name) {
  return (
    name +
    (name[0].toLowerCase() == 'r' ? ' plays' : ' does not play') +
    ' banjo'
  );
}

function areYouPlayingBanjo(name) {
  if (name[0].toLowerCase() === 'r') {
    return name + ' plays banjo';
  } else {
    return name + ' does not play banjo';
  }
}

// test
/*
const chai = require("chai");
const assert = chai.assert;
chai.config.truncateThreshold=0;

describe("Basic tests", () => {
  it("Testing for fixed tests", () => {
    assert.strictEqual(areYouPlayingBanjo("Adam"), "Adam does not play banjo");
    assert.strictEqual(areYouPlayingBanjo("Paul"), "Paul does not play banjo");
    assert.strictEqual(areYouPlayingBanjo("Ringo"), "Ringo plays banjo");
    assert.strictEqual(areYouPlayingBanjo("bravo"), "bravo does not play banjo");
    assert.strictEqual(areYouPlayingBanjo("rolf"), "rolf plays banjo");
    })
  })

*/

File: ifBanjo/if_banjo.py
def are_you_playing_banjo(name):
    '''
    Create a function which answers the question "Are you playing banjo?".
    If your name starts with the letter "R" or lower case "r", you are playing banjo!

    The function takes a name as its only argument, and returns one of the following strings:

        name + " plays banjo" 
        name + " does not play banjo"
        Names given are always valid strings.
    '''
    if name[0].upper() == 'R':
        return f'{name} plays banjo'
    else:
        return f'{name} does not play banjo'

# alternantive solns


def areYouPlayingBanjo(name):
    return name + (' plays' if name[0].lower() == 'r' else ' does not play') + " banjo"


def areYouPlayingBanjo(name):
    return '{} {}'.format(name, 'plays banjo' if name.startswith('R') or name.startswith('r') else 'does not play banjo')

# tests
# import codewars_test as test

# try:
#     from solution import areYouPlayingBanjo as are_you_playing_banjo
# except ImportError:
#     from solution import are_you_playing_banjo

# @test.describe("Fixed Tests")
# def basic_tests():
#     @test.it('Basic Test Cases')
#     def basic_test_cases():
#         test.assert_equals(are_you_playing_banjo("martin"), "martin does not play banjo");
#         test.assert_equals(are_you_playing_banjo("Rikke"), "Rikke plays banjo");
#         test.assert_equals(are_you_playing_banjo("bravo"), "bravo does not play banjo")
#         test.assert_equals(are_you_playing_banjo("rolf"), "rolf plays banjo")
